plugins/woocommerce-blocks/tests/e2e/tests/catalog-sorting/catalog-sorting.block_theme.spec.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { expect, test } from '@woocommerce/e2e-playwright-utils';
+
+const blockData = {
+	name: 'Catalog Sorting',
+	slug: 'woocommerce/catalog-sorting',
+	class: '.wc-block-catalog-sorting',
+};
+
+test.describe( `${ blockData.slug } Block`, () => {
+	test( "block can't be inserted in Post Editor", async ( {
+		editor,
+		admin,
+	} ) => {
+		await admin.createNewPost();
+		await expect(
+			editor.insertBlock( { name: blockData.slug } )
+		).rejects.toThrow(
+			new RegExp( `Block type '${ blockData.slug }' is not registered.` )
+		);
+	} );
+
+	test( 'block can be inserted in Site Editor', async ( {
+		editorUtils,
+		editor,
+		admin,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'woocommerce/woocommerce//archive-product',
+			postType: 'wp_template',
+		} );
+		await editorUtils.enterEditMode();
+		const alreadyPresentBlock = await editorUtils.getBlockByName(
+			blockData.slug
+		);
+		await expect( alreadyPresentBlock ).toHaveText( 'Default sorting' );
```

Comment:
Why are we checking an "already present block" here?

Reply:
> Why are we checking an "already present block" here?

We could remove this check, but why not?

Reply:
The annotation that was just added kinda makes my point! 😅 

<img width="1203" alt="Screenshot 2024-04-29 at 14 09 08" src="https://github.com/woocommerce/woocommerce/assets/1451471/3e5c28b2-e915-4d0b-8d21-7fde23640a42">


Generally, tests should concentrate solely on the target functionality (what they're supposed to test) and proceed in the most direct manner possible. Each additional step increases the test's fragility, potentially causing it to fail even when the functionality it tests remains sound. To use this one as an example, instead of editing the existing template, which required us to remove the existing block before inserting it back, we could create a new template, insert the block and ensure it's there:

```js
const template = await requestUtils.createTemplate( 'wp_template', {
	slug: 'sorter',
	title: 'Sorter',
	content: 'howdy',
} );

await admin.visitSiteEditor( {
	postId: template.id,
	postType: 'wp_template',
} );

await editorUtils.enterEditMode();

await editor.insertBlock( {
	name: 'woocommerce/catalog-sorting',
} );

await expect(
	editor.canvas.getByLabel( 'Block: Catalog Sorting' )
).toBeVisible();
```

What do you think?